index.bs

```diff
@@ -165,37 +165,61 @@ by showing how many sites are willing to ask for intrusive and unnecessary permi
 
 <h3 id="naming-is-hard">Naming things</h3>
 
-Naming is hard! We would all like a silver-bullet for naming APIs...
-
 Names take meaning from:
 
 * signposting (the name itself)
 * use (how people come to understand the name over time)
 * context (the object on the left-hand side, for example)
 
-Consistency is a good principle that helps to create a platform that users can navigate intuitively
-and by name association.
+<h4 id="naming-common-words" class="no-num no-toc">Use common words</h4>
 
-Please consult widely on names in your APIs.
+API naming *must* be done in easily readable US English.
+Keep in mind that most web developers are not native English speakers.
+Whenever possible, names should be chosen that use common vocabulary
+non-native English speakers are likely to understand when first encountering the name.
+Always value readability over brevity.
+
+<p class="example"For example `horizontalAlignment` is a more
+English-readable name than `alignmentHorizontal`.
+
+Note: It may be appropriate to use technical language or well-known terms of art
+in the specification where the API is defined;
+this principle applies only to the API names themselves.
 
-API naming *must* be done in easily readable US English. For example `horizontalAlignment` 
-is a more English-readable name than `alignmentHorizontal`. Whenever possible consider 
-readability over brevity.
+<h4 id="naming-charset" class="no-num no-toc">Use ASCII names</h4>
 
-Names should adhere to the local language restrictions, for example CSS ident rules etc. 
+Names must adhere to the local language restrictions, for example CSS ident rules etc.
 and *should* be in the [ASCII range](https://infra.spec.whatwg.org/#ascii-code-point).
 
+<h4 id="naming-consultation" class="no-num no-toc">Consultation</h4>
+
+Please consult widely on names in your APIs.
```

Comment:
We had some discussion on this in today's teleconference.  One of the issues with this is that it can tend to lead to a lot of bikeshedding.  It may be useful to describe what advice is useful to listen to, e.g., perhaps advice based on underlying principles (perhaps the ones in this document, perhaps more domain-specific ones).

Reply:
I've tried to capture this in 2f9ba01, @atanassov & @dbaron.